### PR TITLE
JS validation fixes for trade B2 and A6

### PR DESCRIPTION
**Pull request by hammerfunk**

### CODE REVIEW

File: app/assets/javascripts/frontend/form-validation.js.coffee [hunk 1]
```diff
@@ -75,17 +75,28 @@ window.FormValidation =
   validateRequiredQuestion: (question) ->
     # if it's a conditional question, but condition was not satisfied
     conditional = true
-    question.find(".js-conditional-question").each () ->
-      if !$(this).hasClass("show-question")
+
+    if question.find(".js-by-trade-goods-and-services-amount").size() > 0
+      # If it's the trade B1 question which has multiple siblings that have js-conditional-question
+      if question.find(".js-conditional-question.show-question .js-by-trade-goods-and-services-amount .js-conditional-question.show-question").size() == 0
```
Comment (matthewford) [inline app/assets/javascripts/frontend/form-validation.js.coffee]: Line exceeds maximum allowed length

File: app/assets/javascripts/frontend/form-validation.js.coffee [hunk 2]
```diff
@@ -75,17 +75,28 @@ window.FormValidation =
...
         conditional = false
+    else
+      question.find(".js-conditional-question").each () ->
```
Comment (matthewford) [inline app/assets/javascripts/frontend/form-validation.js.coffee]: Empty parameter list is forbidden

File: app/assets/javascripts/frontend/form-validation.js.coffee [hunk 3]
```diff
@@ -75,17 +75,28 @@ window.FormValidation =
...
     if !conditional
       return
 
     # This handles questions with multiple fields,
     # like name and address
-    if question.find(".question-group .question-group").length
-      for subquestion in question.find(".question-group .question-group")
+    if question.find(".js-by-trade-goods-and-services-amount").size() > 0
+      # If it's the trade B1 question which has multiple siblings that have js-conditional-question
+      subquestions = question.find(".js-by-trade-goods-and-services-amount .js-conditional-question.show-question .question-group")
```
Comment (matthewford) [inline app/assets/javascripts/frontend/form-validation.js.coffee]: Line exceeds maximum allowed length

**Comment by dreamfall** [comment]
Nice  :shipit:

**Pull request merged**